tests/integration/pass/strings.ncl

```diff
@@ -11,6 +11,123 @@ let {check, ..} = import "lib/assert.ncl" in
     "Hello, %{x}! Welcome in %{let y = "universe" in "the %{x}-%{y}"}")
     == "Hello, world! Welcome in the world-universe",
 
+  # happy path for all stdlib functions
+  # string.BoolLiteral
+  ("True" | string.BoolLiteral) == "true",
+  ("False" | string.BoolLiteral) == "false",
+  ("true" | string.BoolLiteral) == "true",
+  ("false" | string.BoolLiteral) == "false",
+  # string.NumLiteral
+  ("+1.2" | string.NumLiteral) == "+1.2",
+  ("-1.2" | string.NumLiteral) == "-1.2",
+  ("9001" | string.NumLiteral) == "9001",
+  # string.CharLiteral
+  ("e" | string.CharLiteral) == "e",
+  # string.EnumTag
+  (`Foo | string.EnumTag) == `Foo,
+  # string.Stringable
+  (`Foo | string.Stringable) == `Foo,
+  (true | string.Stringable) == true,
+  (1 | string.Stringable) == 1,
+  ("_" | string.NonEmpty) == "_",
+  # string.join
+  string.join " " ["Hello", "世界"] == "Hello 世界",
+  string.join "anything" [] == "",
+  string.join "a" ["b", "n", "n", ""] == "banana",
+  # string.split
+  string.split "," "1,2,3" == ["1", "2", "3"],
+  string.split "" "123" == ["", "1", "2", "3", ""],
+  string.split "❤" "👩‍❤️‍💋‍👨" == [ "👩‍", "️‍💋‍👨" ],
```

Comment:
I'm not sure if this one is correct or not. It seems arguable either way. Same for the one below this one.